cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -432,7 +446,35 @@ if(NOT DEFINED CUDF_VERSION_Arrow)
   )
 endif()
 
+# If the pyarrow package contains libarrow we want to link against it directly instead of searching
```

Comment:
This logic should be explicitly opt-in.
If you are building libcudf on a machine with a system installed pyarrow you would always use that with no way to opt out.

Reply:
Ugh you're right, but this is unfortunate. It does mean we continue to have to specify nonzero information from the CLI when invoking certain (wheel) builds, but I guess that's not the worst.